drivers/ethernet/eth_stm32_hal.c

```diff
@@ -563,39 +563,39 @@ static struct net_pkt *eth_rx(const struct device *dev, uint16_t *vlan_tag)
 	}
 #endif /* CONFIG_SOC_SERIES_STM32H7X */
 
+	if (pkt) {
```

Comment:
If you do
```
if (!pkt) {
	goto out;
}
.....
out:
	if (!pkt) {
		eth_stats_update_errors_rx(get_iface(dev_data, *vlan_tag));
	}

	return pkt;
}
```
Then we do not have to have hard to read if() else() branches.

Reply:
Good idea, updated.